src/DGtal/images/ArrayImageAdapter.h

```diff
@@ -0,0 +1,564 @@
+   *
+   * Some helpers are available (see makeArrayImageAdapterFromIterator and makeArrayImageAdapterFromImage)
+   * for easy construction (with template deduction) from an iterator or an CConstImage model.
+   *
+   * It is important to note that, since this class only adapts an already existing storage to an image, all 
+   * copy operations (constructor and operator) lead to shallow copies.
+   *
+   * The following code snippet demonstrates how to use ArrayImageAdapter from converting a native C-array to an image:
+   * @code
+   * using Space = SpaceND<2>;
+   * using Domain = HyperRectDomain<Space>;
+   * using Point = typename Space::Point;
+   * using Value = double;
+   *
+   * const Domain domain{ {0, 1}, {4, 3} };
+   *
+   * Value* data = new Value[ domain.size() ];
+   *
+   * // Convert this allocated memory to a CImage model. 
+   * ArrayImageAdapter< Value*, Domain > image( data, domain );
+   * // Alternative syntax using the helpers:
+   * // auto image = makeArrayImageAdapterFromIterator( data, domain );
+   * 
+   * // Fill the image with first coordinate of the point
+   * for ( auto it = image.begin(); it != image.end(); ++it )
+   *   {
+   *     *it = it.getPoint()[0];
+   *   }
+   *
+   * // Get a constant view on a sub-domain.
+   * const Domain sub_domain{ {1, 1}, {3, 2} };
+   * ArrayImageAdapter< Value const*, Domain > cst_image( data, domain, sub_domain );
+   * // Alternative syntax using the helpers:
+   * // auto cst_image = makeArrayImageAdapterFromImage( image, sub_domain );
+   *
+   * // Display it.
+   * for ( auto value : cst_image )
+   *   {
+   *     std::cout << value << " ";
+   *   }
+   * std::cout << std::endl;
+   *
+   * @endcode
+   *
+   * @remark The given random-access iterator can be either mutable or constant.
+   *
+   * @warning The array must be column-major ordered (but row-major order could be later accepted via template parameter, if needed ?)
+   * @warning The domain must be an HyperRectDomain.
+   * @warning C++11 needs to be enabled in order to use this class.
+   *
+   * @tparam TArrayIterator Type of a random-access iterator over the datas (can be a T* pointer).
+   * @tparam TSpace Type of the space associated to the HyperRectDomain (auto-deduced from TDomain template, see ArrayImageAdapter).
+   *
+   * @see makeArrayImageAdapterFromIterator
+   * @see makeArrayImageAdapterFromImage
+   */
+  template <
+    typename TArrayIterator,
+    typename TSpace
+  >
+  class ArrayImageAdapter< TArrayIterator, HyperRectDomain<TSpace> >
+      : public IteratorCompletion< ArrayImageAdapter< TArrayIterator, HyperRectDomain<TSpace> > >
+    {
+
+    // Checks Random-access iterator concept on TArrayIterator
+    BOOST_CONCEPT_ASSERT( (boost_concepts::RandomAccessTraversalConcept<TArrayIterator>) );
+
+    public:
+      // Typedefs
+      using Self = ArrayImageAdapter<TArrayIterator, HyperRectDomain<TSpace> >;
```

Comment:
if it is, please switch to ```typedef```, all inner types in dgtal  use this syntax (with associated doxygen doc).